src/test/storage/chunk_test.cpp

```diff
@@ -58,4 +59,35 @@ TEST_F(StorageChunkTest, RetrieveSegment) {
   EXPECT_EQ(base_segment->size(), 4u);
 }
 
+TEST_F(StorageChunkTest, ApplyDictionaryEncoding) {
```

Comment:
Additionally testing for adding an invalid segment (in this case for example having more than 3 values) could be beneficial